app/components/gh-image-uploader.js

```diff
@@ -177,7 +192,9 @@ export default Component.extend({
     generateRequest() {
         let ajax = this.get('ajax');
         let formData = this.get('formData');
-        let url = `${ghostPaths().apiRoot}/uploads/`;
+        let isIcon = this.get('isIcon');
+        // CASE: we want to upload an icon and we have to POST it to a different endpoint, expecially for icons
+        let url = isIcon ? `${ghostPaths().apiRoot}/uploads/icons/` : `${ghostPaths().apiRoot}/uploads/`;
```

Comment:
After going through the existing code in core and having a conversation with @kirrg001, we decided to post icons to a new endpoint. Therefore, I created the property `isIcon` which checks in the `_defaultValidator` if the extensions with which `gh-image-uploader` is called include the `ico` extension. In this case, we have an icon for sure and I set the correlated property to `true`.